integration/plugin/authz/authz_plugin_test.go

```diff
@@ -176,15 +187,17 @@ func TestAuthZPluginAPIDenyResponse(t *testing.T) {
 	daemonURL, err := url.Parse(d.Sock())
 	assert.NilError(t, err)
 
-	conn, err := net.DialTimeout(daemonURL.Scheme, daemonURL.Path, time.Second*10)
+	client, err := socketHTTPClient(daemonURL)
 	assert.NilError(t, err)
-	c := httputil.NewClientConn(conn, nil)
-	req, err := http.NewRequest(http.MethodGet, "/version", nil)
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, "/version", nil)
 	assert.NilError(t, err)
-	req = req.WithContext(ctx)
-	resp, err := c.Do(req)
+	req.URL.Scheme = "http"
+	req.URL.Host = daemonURL.Path
```

Comment:
So we need to use `client.DummyHost` here instead? (see https://github.com/moby/moby/pull/45942)

```suggestion
	req.URL.Host = client.DummyHost
```

Reply:
I don't think our integration tests need to be compatible with Go 1.20.6, 1.20.7 or 1.21.0. It was a regression, and has been fixed since Go 1.20.8, 1.21.1. We don't have to work around it any longer.

Reply:
`DummyHost` is maybe clearer, at least it would be much better for constructing a full URL. Without setting any of this it fails and setting the Host to path is still weird with the slashes encoded.